src/psl/psl-node.h

```diff
@@ -108,6 +115,17 @@ ident_t psl_ident(psl_node_t p);
 bool psl_has_ident(psl_node_t p);
 void psl_set_ident(psl_node_t p, ident_t i);
 
+unsigned psl_ranges(psl_node_t p);
+tree_t psl_range(psl_node_t p, unsigned n);
+void psl_add_range(psl_node_t p, tree_t r);
```

Comment:
Let's instead make this `psl_repeat` as there is only ever zero or one rather than a list (add an `I_REPEAT` to to `src/object.h`).

I think we could also have a new `P_REPEAT` node that has a `psl_subkind` and a `psl_tree` for the count/range, and `P_SERE` has an optional `psl_repeat`.

Reply:
OK, I see, that makes sense. I was thinking about this "redundancy". I first wanted to use the array to store multiple consecutive repetitions, e.g. in case like: `a[*2][*4][*6]`, and keep a single `P_SERE` object. However, then I realized
that `Proc_Block` can arbitrarily alterate with the "repeat", so then there is no way to track what all should be repeated.

So, the `P_REPEAT` will have `value` that will point to:
- `P_SERE` in case of braced/clocked SERE.
- `P_HDL_EXPR` in case of boolean.

Reply:
I was actually thinking it would work the other way round, so that `P_SERE` would hold the `P_REPEAT`.

```
a[*2][*4][*6]  ==  {{{a}[*2]}[*4]}[*6]

P_SERE
  \-----repeat----P_REPEAT---subkind--PSL_CONSEC_REPEAT
  |                   |
  |                   \-----value----T_LITERAL(6)
  |
  \-----operand[0]----P_SERE
                         \-----repeat----P_REPEAT----subkind--PSL_CONSEC_REPEAT
                         |                   |
                         |                   \------value---T_LITERAL(4)
                         |
                         \-----operand[0]----P_SERE (etc.)
```

Reply:
OK that makes sense, the first object encountered during tree traversal is then always `P_SERE`.

Reply:
But still the `P_REPEAT` needs at least `left` and `right`, since repeat can be range (instead of plain `value`).

Reply:
You can just use a `T_RANGE` tree object and assign it to the value.